app/services/user_signup.rb

```diff
@@ -0,0 +1,47 @@
+class UserSignup
+  attr_reader :error_message
+  
+  def initialize(user)
+    @user = user
+  end
+  
+  def sign_up(stripe_token, invitation_token)
+    if @user.valid?                 
+      charge = StripeWrapper::Charge.create(
+        amount: 999,
+        card: stripe_token,
+        description: "MyFlix Sign Up Charge for #{@user.email}"
+      )
+      if charge.successful?
+        @user.save     
+        handle_invitation(invitation_token)
+        AppMailer.delay.send_welcome_email(@user)
+        @status = :success
+        self
+      else
+        @status = :fail
+        @error_message = charge.error_message
+        self
+      end
+    else
+      @status = :fail
+      @error_message = "Invalid user information. Please check error messages below."
+      self
```

Comment:
You could refactor `self` at the end of this method, because all `if` branches will end up returning it.